ssz/sedes/vector.py

```diff
@@ -0,0 +1,89 @@
+from typing import (
+    Generator,
+    Sequence,
+    Tuple,
+    TypeVar,
+)
+
+from eth_utils import (
+    to_tuple,
+)
+
+from ssz.exceptions import (
+    DeserializationError,
+    SerializationError,
+)
+from ssz.sedes.base import (
+    BaseSedes,
+    CompositeSedes,
+)
+
+TSerializableElement = TypeVar("TSerializable")
+TDeserializedElement = TypeVar("TDeserialized")
+
+
+class Vector(CompositeSedes[Sequence[TSerializableElement], Tuple[TDeserializedElement, ...]]):
+
+    def __init__(self,
+                 number_of_elements: int,
+                 element_sedes: BaseSedes[TSerializableElement, TDeserializedElement]) -> None:
```

Comment:
Maybe reorder to `element_sedes` and then `number_of_elements`? The notation in the spec is `[type, N]`, e.g., `'latest_randao_mixes': ['bytes32', LATEST_RANDAO_MIXES_LENGTH]`.

Reply:
Good point. I actually ran into this a couple of times when writing the test generator.

As this changes the interface anyway, I'm going to rename `number_of_elements` to `length` as well (to be consistent terminology-wise).